src/inline.d

```diff
@@ -1523,6 +1523,35 @@ public:
                 }
             }
         }
+        else if (e.e1.op == TOKstar &&
+                 (cast(PtrExp)e.e1).e1.op == TOKvar)
+        {
+            /* See if calling function pointer, and that function pointer is only
+             * assigned its _init.
+             * If so, do 'copy propagation' of the _init value and try to inline it.
+             */
+            VarExp ve = cast(VarExp)(cast(PtrExp)e.e1).e1;
+            VarDeclaration v = ve.var.isVarDeclaration();
+            if (v && v._init && isHeadConst(v, parent))
+            {
+                //printf("init: %s\n", v._init.toChars());
+                auto ei = v._init.isExpInitializer();
+                if (ei && ei.exp.op == TOKblit)
+                {
+                    Expression e2 = (cast(AssignExp)ei.exp).e2;
+                    if (e2.op == TOKfunction)
+                    {
+                        auto fld = (cast(FuncExp)e2).fd;
```

Comment:
I'll be adding many more cases. Once that's done, I'll look for ways to refactor it.